apps/pipelines/nodes/nodes.py

```diff
@@ -984,6 +984,9 @@ def _get_custom_globals(self, state: PipelineState):
                 "set_temp_state_key": self._set_temp_state_key(state),
                 "get_session_state_key": self._get_session_state_key(state["experiment_session"]),
                 "set_session_state_key": self._set_session_state_key(state["experiment_session"]),
+                "get_route": lambda node_name: state.get_route(node_name),
+                "get_node_path": lambda node_name: state.get_node_path(node_name),
+                "get_all_routes": lambda: state.get_all_routes(),
```

Comment:
I think 'state' here is just a dict - if you want access to these functions you'll need to wrap it: `state = PipelineState(**state)` (untested).

Also, you can drop the lambdas:
```suggestion
                "get_route": state.get_route,
                "get_node_path": state.get_node_path,
                "get_all_routes": state.get_all_routes,
```